java/src/main/java/io/opentracing/grpc/ServerTracingInterceptor.java

```diff
@@ -0,0 +1,234 @@
+    ) {
+        Map<String, String> headerMap = new HashMap<String, String>();
+        for (String key : headers.keys()) {
+            String value = headers.get(Metadata.Key.of(key, Metadata.ASCII_STRING_MARSHALLER));
+            headerMap.put(key, value);
+        }
+
+        final String operationName;
+        if (this.operationName.equals("")) {
+            operationName = call.getMethodDescriptor().getFullMethodName();            
+        } else {
+            operationName = this.operationName;
+        }
+        final Span span = getSpanFromHeaders(headerMap, operationName);
+
+        for (ServerRequestAttribute attr : this.tracedAttributes) {
+            switch (attr) {
+                case METHOD_TYPE:
+                    span.setTag("Method Type", call.getMethodDescriptor().getType().toString());
+                    break;
+                case METHOD_NAME:
+                    span.setTag("Method Name", call.getMethodDescriptor().getFullMethodName());
+                    break;
+                case CALL_ATTRIBUTES:
+                    span.setTag("Call Attributes", call.attributes().toString());
+                    break;
+                case HEADERS:
+                    span.setTag("Headers", headers.toString());
+                    break;
+            }
+        }
+
+        Context ctxWithSpan = Context.current().withValue(OpenTracingContextKey.getKey(), span);
+        ServerCall.Listener<ReqT> listenerWithContext = Contexts
+            .interceptCall(ctxWithSpan, call, headers, next);
+
+        ServerCall.Listener<ReqT> tracingListenerWithContext = 
+            new ForwardingServerCallListener.SimpleForwardingServerCallListener<ReqT>(listenerWithContext) {
+
+            @Override
+            public void onMessage(ReqT message) {
+                if (streaming || verbose) { span.log("Message received", message); }
+                delegate().onMessage(message);
+            }
+
+            @Override 
+            public void onHalfClose() {
+                if (streaming) { span.log("Client finished sending messages", null); }
+                delegate().onHalfClose();
+            }
+
+            @Override
+            public void onCancel() {
+                span.log("Call cancelled", null);
+                span.finish();
+                delegate().onCancel();
+            }
+
+            @Override
+            public void onComplete() {
+                if (verbose) { span.log("Call completed", null); }
+                span.finish();
+                delegate().onComplete();
+            }
+        };
+
+        return tracingListenerWithContext; 
+    }
+
+    private Span getSpanFromHeaders(Map<String, String> headers, String operationName) {
+        Span span;
+        try {
+            SpanContext parentSpanCtx = tracer.extract(Format.Builtin.HTTP_HEADERS, 
+                new TextMapExtractAdapter(headers));
+            if (parentSpanCtx == null) {
```

Comment:
There doesn't seem to be anything in the Java API (addChildOf [here](https://github.com/opentracing/opentracing-java/blob/master/opentracing-api/src/main/java/io/opentracing/Tracer.java#L99)) that says that null references should be ignored, or is that just implied?